phpBB/phpbb/style/manager.php

```diff
@@ -0,0 +1,436 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\style;
+
+use phpbb\style\exception;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+* The extension manager provides means to activate/deactivate extensions.
+*/
+class manager
+{
+	protected $cache;
+
+	/** @var ContainerInterface */
+	protected $container;
+
+	protected $db;
+	protected $config;
+	protected $filesystem;
+	protected $styles_table;
+	protected $users_table;
+	protected $phpbb_root_path;
+
+	/** @var \phpbb\textformatter\cache_interface */
+	protected $text_formatter_cache;
+
+	protected $reserved_style_names = array('adm', 'admin', 'all');
+
+	protected $styles_path;
+	protected $styles_path_absolute = 'styles';
+
+	public function __construct(\phpbb\cache\service $cache = null, \phpbb\config\config $config, ContainerInterface $container, \phpbb\db\driver\driver_interface $db, \phpbb\filesystem\filesystem_interface $filesystem, $styles_table, $users_table, $phpbb_root_path)
+	{
+		$this->cache = $cache;
+		$this->config = $config;
+		$this->container = $container;
+		$this->db = $db;
+		$this->filesystem = $filesystem;
+		$this->styles_table = $styles_table;
+		$this->users_table = $users_table;
+		$this->phpbb_root_path = $phpbb_root_path;
+
+		$this->text_formatter_cache = $container->get('text_formatter.cache');
```

Comment:
This was going to be moved to a listener:
https://github.com/phpbb/phpbb/pull/4789/files#diff-fa5782159e6d410543f46209e2424206R122